lib/virt.py

```diff
@@ -403,19 +370,33 @@ def install(self, location=None, kickstart=None, ks_verbatim=False):
         if not kickstart:
             kickstart = Kickstart()
 
+        http_port = 8090
+
         if not ks_verbatim:
             kickstart.add_host_repos()
-            kickstart.add_firstboot_setup()
             util.ssh_keygen(self.ssh_keyfile_path)
             with open(f'{self.ssh_keyfile_path}.pub') as f:
                 pubkey = f.read().rstrip()
             kickstart.add_authorized_key(pubkey)
+            kickstart.add_install_only_repo('rpmpack', f'http://{NETWORK_HOST}:{http_port}/repo')
+            kickstart.add_packages([util.RPMPACK_NAME])
 
         disk_path = f'{GUEST_IMG_DIR}/{self.name}.img'
         disk_format = 'raw'
-
         cpus = os.cpu_count() or 1
-        with kickstart.to_tmpfile() as ksfile:
+
+        pack = util.RpmPack()
+        pack.post.append(GUEST_SETUP)
+        pack.requires += GUEST_SETUP_REQUIRES
+
+        srv = util.BackgroundHTTPServer(NETWORK_HOST, http_port)
+
+        with contextlib.ExitStack() as stack:
+            repo = stack.enter_context(pack.build_as_repo())
+            srv.add_file(repo, 'repo')
+            stack.enter_context(srv)
+            ksfile = stack.enter_context(kickstart.to_tmpfile())
```

Comment:
Smart.
However, documentation is missing - in README, in function doc, or just a comment is fine.

Reply:
I can add commented sections for each of `RpmPack`, `Kickstart`, and more if needed, but I probably wouldn't explain the behavior of standard Python modules - ie. we don't explain `NamedTuple` or ie. `ChainMap` either.

Reply:
Yes, definitely don't explain those, everyone can google that (as I did with the `ExitStack` :D). All I want documented is the idea of package added to installation which does the setup to allow login.